sd/etcd/client_test.go

```diff
@@ -135,7 +135,7 @@ func (fw *fakeWatcher) Next(context.Context) (*etcd.Response, error) {
 			return nil, nil
 		case <-fw.err:
 			return nil, errors.New("error from underlying etcd watcher")
-		default:
+
```

Comment:
This results in different behavior, does it not?

Reply:
It doesn't change the observed behavior (it'll still block until either one of the first two cases return something).

TBH I don't really understand why the for loop is necessary.

Reply:
Well if you have the default, you need the for, otherwise this is a one-and-done.

But it's still a terrible pattern, as it turns what could be a simple select into a hot loop?? Maybe an artifact from refactoring.

Reply:
Probably, yeah. @withshubh could you please remove the for loop as well?